### PR TITLE
Fixes #6468 - Allow filtering/searching by active user

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@tbrisker looks good on first glance, but would require some unit/functional tests please :)

**Comment by tbrisker** [comment]
Added tests, and also aliased owner to user.login and improved searching.

File: app/models/concerns/hostext/search.rb [hunk 1]
```diff
@@ -81,7 +81,14 @@ module Search
...
       def search_by_user(key, operator, value)
-        key_name = User.connection.quote_column_name(key.sub(/^.*\./,''))
+        clean_key = key.sub(/^.*\./,'')
+        if value == "Active_User"
+          value = User.current.id
+          clean_key = "id"
+        elsif key == "owner"
+          clean_key = "login"
+        end
+        key_name = User.connection.quote_column_name(clean_key)
```
Comment (tbrisker) [inline app/models/concerns/hostext/search.rb:91]: Turns out using ` for quoting column names works only on mysql/sqlite - which caused the tests to break on postgesql. 
Using User.connection.quote_column_name(...) quotes the name correctly for each db and now tests pass.

**Comment by isratrade** [comment]
@tbrisker, I can an error when I test this locally

```
ArgumentError (wrong number of arguments (2 for 1)):
  app/models/host/managed.rb:23:in `complete_for'
```

**Comment by isratrade** [comment]
looks good after changing 'Active_user' to 'current_user'

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -20,6 +20,13 @@ class Host::Managed < Host::Base
 
   has_one :token, :foreign_key => :host_id, :dependent => :destroy
 
+  def self.complete_for(query, opts = {})
+    matcher = /(((user\.[a-z]+)|owner)\s*[=~])\s*\S+\s*\z/
+    output = super(query)
+    output << output.last.sub(matcher,'\1 current_user') if not output.empty? and output.last =~ matcher
```
Comment (isratrade) [inline app/models/host/managed.rb:26]: lgtm except that 'current_user' appears at the bottom of the list. Maybe it's better at the top of the list or in alphabetical order. Also, the autocomplete doesn't work when typing 'cu..." However, when typing in something else (ex. bob ...) then 'current_user' still appears in the list, and I'm not sure it should. I feel ok to merge as is and create these as a separate issue. http://projects.theforeman.org/issues/7440

**Comment by isratrade** [comment]
merged as 5dfb51bbe0b7c1ff91868437f627e6c535d87542. thanks @tbrisker !

**Pull request closed without merging**